### DISCUSSION TITLE
Cannot edit pages

### DISCUSSION THREAD

**Discussion by mrloldev** [outline/outline#3546] [Self Hosting] [answered]
**To Reproduce**
1. Create wiki from source
2. Build it

**Expected behavior**
I think clicking in the body or pressing `e` I could edit the page however I can't edit nothing.

**Outline (please complete the following information):**
- Install: 0.63.0

**Desktop (please complete the following information):**
 - OS: Ubuntu
 - Version 20.04

**Comment by mrloldev** [comment]
**To Reproduce**
1. I also use pm2 and nginx

**Reply by Limezy** [reply]
So everything works except the edit ? 
Is collaborative editing on or off ?

**Reply by mrloldev** [reply]
it is on

**Reply by mrloldev** [reply]
I tryed to turn off but I can not so I think it wasn't this the problem. It is?

**Comment by mrloldev** [comment]
The solution was disabling the collaborative editing to do this I didn't follow #3388 What I do was going to the folder server/models/Team then changing the default value of collaborativeEditing from true to false.

**Comment by ksaadDE** [comment]
I solved the same problem having a nginx reverse proxy.

I just edited the server block nginx conf file and added:
```
location /
{
....
      proxy_http_version 1.1;
      proxy_set_header Upgrade $http_upgrade;
      proxy_set_header Connection "upgrade";
}
```

At the end of the location block that has the proxy_pass directive (it must be after proxy_pass - whyever)

**Reply by timvdhoorn** [reply]
Thanks for this! This worked!

**Reply by Bibowa** [reply]
I have the same issue, but with NPM if you write this rules using custom locations, the config generated file will have the proxy_pass at the bottom of the configuration (as you can see from the file).

 ```
 location / {
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection "Upgrade";
    proxy_set_header Host $host;

    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;proxy_set_header Host $host;
    proxy_set_header Host $http_host;
    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header X-Scheme $scheme;
    proxy_set_header X-Forwarded-Proto $scheme;
    proxy_redirect off;

    proxy_set_header Host $host;
    proxy_set_header X-Forwarded-Scheme $scheme;
    proxy_set_header X-Forwarded-Proto  $scheme;
    proxy_set_header X-Forwarded-For    $remote_addr;
    proxy_set_header X-Real-IP          $remote_addr;
    proxy_pass       ip_and_port
```


 How can i solve this?

**Reply by DebrodeD** [reply]
I had this issue as well (running NPM & Outline & Keycloak as containers on my unraid server). I had to set a custom location in order to use the "set header" directives in NPM (the GUI). Finally I found that removing all the extra directives on the outline container Proxy Host in NPM and just using "proxy_set_header X-Forwarded-Host $host;" worked for me (with location set to "/").

**Reply by mayknxyz** [reply]
> I solved the same problem having a nginx reverse proxy.
> 
> I just edited the server block nginx conf file and added:
> 
> ```
> location /
> {
> ....
>       proxy_http_version 1.1;
>       proxy_set_header Upgrade $http_upgrade;
>       proxy_set_header Connection "upgrade";
> }
> ```
> 
> At the end of the location block that has the proxy_pass directive (it must be after proxy_pass - whyever)

This worked for me! v0.83.0

**Reply by rzm25** [reply]
> ```
> location /
> {
> ....
>       proxy_http_version 1.1;
>       proxy_set_header Upgrade $http_upgrade;
>       proxy_set_header Connection "upgrade";
> }
> ```

I don't suppose anyone knows what this code would look like in Caddy? I'm having the same issue

**Reply by baselinesKevin** [reply]
I did eventually get this working.

Here's what I did

```
outline.host.com {
        @ws {
                header Connection *Upgrade*
                header Upgrade websocket
        }
        reverse_proxy localhost:8500 {
                transport http {

                }
                
        }
}
```

**Reply by tommoor** [reply, marked as answer]
The requirements for nginx have long been documented here btw.

[Outline Nginx config](https://docs.getoutline.com/s/hosting/doc/nginx-6htaRboR57)

**Comment by baselinesKevin** [comment]
I'm using Caddy which is supposed to just work for Websockets. Do I need to enable an extra port for docker just for Websockets?

**Reply by tommoor** [reply]
Caddy works with websockets out of the box, so your issue likely is somewhere else

**Reply by baselinesKevin** [reply]
Maybe, but  I had to add the at WS directive which apparently messes up code formatting here on Github -- but you could follow that and it would work. For me it didn't work until I did the header Connection Upgrade bit and called out ws specifically.

**Reply by rzm25** [reply]
> Maybe, but I had to add the at WS directive which apparently messes up code formatting here on Github -- but you could follow that and it would work. For me it didn't work until I did the header Connection Upgrade bit and called out ws specifically.

Try putting three ` ` ` instead of one

**Reply by baselinesKevin** [reply]
done!
